applicationinsights/channel/AsynchronousSender.py

```diff
@@ -123,6 +129,14 @@ def _run(self):
                 # if we didn't get any items from the queue, we're done here
                 if len(data) == 0:
                     break
+                
+                # Drop extra events from the queue until it no larger than send_buffer_size.
+                # This is a necessary evil to prevent the backlog from bloating forever in bandwidth-limited environments
+                eventsToDrop = max(0, local_queue._queue.qsize() - self._send_buffer_size)
```

Comment:
Silently dropping data that's larger than the buffer size doesn't seem like a good solution. What if you are not bandwidth limited but merely had a burst of events? I would expect the SDK to send this telemetry out over time, not drop it. Users of the SDK generating more telemetry than they can send out of the pipe should reduce the amount of telemetry they send (eg. reducing logging, sampling)